src/storage/mod.rs

```diff
@@ -137,25 +154,47 @@ impl Table {
             println!("Magic Number not correct");
             return Err(DatabaseError::MagicNmbr)
         }
-        let data: Table = try!(decode_from(&mut file, SizeLimit::Infinite));
-        println!("{:?}", data);
-        Ok(())
+        let table: Table = try!(decode_from(&mut file, SizeLimit::Infinite));
+        println!("{:?}", table);
+
+        Ok(table)
     }
+}
 
-    pub fn create_new(engine: u8, name: &str) -> Table {
+/// Table struct that contains the table information
+#[derive(Debug, RustcDecodable, RustcEncodable)]
+pub struct Table {
+    version_nmbr: u8,
+    engine_id: u8,
+    columns: Vec<Column>,
+    name: String,
+    name_of_database: String,
```

Comment:
Would prefer the name `database_name`. Names are really important, one should take some time to think about a good and short name. The importance of names in source code is often underappreciated...